--- /usr/share/vim/vim71/macros/less.vim	2008-07-21 22:46:41.000000000 +0900
+++ /home/ywtnb/.vim/macros/less.vim	2008-12-24 18:39:52.000000000 +0900
@@ -1,6 +1,7 @@
 " Vim script to work like "less"
 " Maintainer:	Bram Moolenaar <Bram@vim.org>
-" Last Change:	2006 Dec 05
+" Modifier:     Watanabe Yuki <magicant.starmen@nifty.com>
+" Last Change:	2007 Dec 07
 
 " Avoid loading this file twice, allow the user to define his own script.
 if exists("loaded_less")
@@ -83,23 +84,24 @@
 endfun
 
 " Scroll one page forward
-noremap <script> <Space> :call <SID>NextPage()<CR><SID>L
+noremap <script> <Space> <C-F><SID>L
+" noremap <script> <Space> :call <SID>NextPage()<CR><SID>L
 map <C-V> <Space>
 map f <Space>
 map <C-F> <Space>
 map z <Space>
 map <Esc><Space> <Space>
-fun! s:NextPage()
-  if line(".") == line("$")
-    if argidx() + 1 >= argc()
-      quit
-    endif
-    next
-    1
-  else
-    exe "normal! \<C-F>"
-  endif
-endfun
+"fun! s:NextPage()
+"  if line(".") == line("$")
+"    if argidx() + 1 >= argc()
+"      quit
+"    endif
+"    next
+"    1
+"  else
+"    exe "normal! \<C-F>"
+"  endif
+"endfun
 
 " Re-read file and page forward "tail -f"
 map F :e<CR>G<SID>L:sleep 1<CR>F
@@ -168,7 +170,7 @@
   else
     noremap <script> N Hg0Nzt<SID>L
   endif
-  cnoremap <silent> <script> <CR> <CR>:cunmap <lt>CR><CR>zt<SID>L
+  cnoremap <script> <CR> <CR>:cunmap <lt>CR><CR>zt<SID>L
 endfun
 
 fun! s:Backward()
@@ -179,7 +181,7 @@
     noremap <script> n Hg0nzt<SID>L
   endif
   noremap <script> N H$Nzt<SID>L
-  cnoremap <silent> <script> <CR> <CR>:cunmap <lt>CR><CR>zt<SID>L
+  cnoremap <script> <CR> <CR>:cunmap <lt>CR><CR>zt<SID>L
 endfun
 
 call s:Forward()
